rust/automerge/src/iter/map_range.rs

```diff
@@ -95,38 +116,185 @@ struct MapIter<'a> {
 impl Shiftable for MapIter<'_> {
     fn shift_next(&mut self, range: Range<usize>) -> Option<<Self as Iterator>::Item> {
         let id = self.id.shift_next(range.clone());
-        let key_str = self.key_str.shift_next(range.clone());
+        let key = self.key_str.shift_next(range.clone());
         let action = self.action.shift_next(range.clone());
         let value = self.value.shift_next(range);
         let pos = self.id.pos() - 1;
-        Some((key_str??, action?, value?, id?, pos))
+        Some(Map {
+            key: key??,
+            action: action?,
+            value: value?,
+            id: id?,
+            pos,
+        })
+    }
+}
+
+#[derive(Clone, Debug)]
+struct Map<'a> {
```

Comment:
`Map` feels like the wrong name for this type because it's not the whole map, it's more like the operation which created a map entry right? Something like `MapOp` or `ResolvedMapOp` or `MapEntry`?